### DISCUSSION TITLE
One time Fetching of uery by range.

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#5315] [Q&A]
Hello, we have a use case of a rolling timeline where a user can with a  mouse scroll focus and un-focus in a timeline.
Scrolling can increase/decrease the interval and show between 15 minutes-1 hour time frame from 1- to 7 days. 

As the user scrolls the timeline we want to fech evens from a backend service. the problem is that there are thousands of events and the query is a bit heavy so we would like to not fetch the same time frame again if it has been fetched regardless of the timeline range.

Eg. if the user is currently viewing 1PM-2PM we want to fetch those events. if then the user expands the view to include 12-4pm we want to only fetch 12-1 and 2-4 . basically always fetching the hours that are hasnt been fetch yet...

Any idea of how we can start approaching this?

**Comment by TkDodo** [comment]
I'd say it depends on how your API is structured. If it allows for batching, this could be a good approach: 

- https://github.com/TanStack/query/discussions/365#discussioncomment-1867860

**Reply by cubecleveland** [reply]
HI TkDodo,

My API will not support that. i have to pass 1 range of dates per api call.
Can i somehow use useQueries ? how would i then pass only the hours i need in the function key ?

Thanks for your help.
